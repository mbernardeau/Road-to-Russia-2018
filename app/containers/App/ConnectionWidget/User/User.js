import React, {
  PureComponent,
} from 'react';
import { pathToJS, firebaseConnect } from 'react-redux-firebase';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Avatar from 'material-ui/Avatar';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import styles from './User.scss';

@firebaseConnect()
@connect(
  // Map state to props
  ({ firebase }) => ({
    user: pathToJS(firebase, 'profile'),
  })
)
class User extends PureComponent { // eslint-disable-line react/prefer-stateless-function
  static propTypes = {
    user: PropTypes.shape({
      avatarUrl: PropTypes.string.isRequired,
      displayName: PropTypes.string.isRequired,
    }).isRequired,
    firebase: PropTypes.shape({
      logout: PropTypes.func.isRequired,
    }).isRequired,
  };

  render() {
    const { user, firebase } = this.props;
    const { logout } = firebase;

    return (
      <div className={styles.userContainer}>
        <Avatar src={user.avatarUrl} style={{ marginRight: 10 }} />

        <span className={styles.username}>{user.displayName}</span>

        <IconMenu
          iconButtonElement={<IconButton><MoreVertIcon color="white" /></IconButton>}
          anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
          targetOrigin={{ horizontal: 'right', vertical: 'top' }}
        >
          <MenuItem primaryText="Se dÃ©connecter" onClick={logout} />
        </IconMenu>
      </div>
    );
  }
}

export default User;
