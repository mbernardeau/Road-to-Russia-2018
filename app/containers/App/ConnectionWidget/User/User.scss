.userContainer {
  display: flex;
  align-items: center;
}

.username {
  color: white;
  font-size: 18px;
}