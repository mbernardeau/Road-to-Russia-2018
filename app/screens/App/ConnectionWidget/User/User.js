import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Radium from 'radium'

import Avatar from 'material-ui/Avatar'
import IconButton from 'material-ui/IconButton'
import Menu, { MenuItem } from 'material-ui/Menu'
import MoreVertIcon from 'material-ui-icons/MoreVert'

class User extends PureComponent {
  // eslint-disable-line react/prefer-stateless-function
  static propTypes = {
    user: PropTypes.shape({
      avatarUrl: PropTypes.string.isRequired,
      displayName: PropTypes.string.isRequired,
    }).isRequired,
    firebase: PropTypes.shape({
      logout: PropTypes.func.isRequired,
    }).isRequired,
  }

  state = {
    anchorEl: null,
    open: false,
  }

  handleClick = event => {
    this.setState({ open: true, anchorEl: event.currentTarget })
  }

  handleRequestClose = () => {
    this.setState({ open: false })
  }

  render() {
    const { user, firebase } = this.props
    const { logout } = firebase

    return (
      <div style={styles.user}>
        <Avatar src={user.avatarUrl} style={styles.avatar} />

        <span style={styles.username}>{user.displayName}</span>

        <IconButton
          aria-label="Plus"
          aria-owns={this.state.open ? 'long-menu' : null}
          aria-haspopup="true"
          onClick={this.handleClick}
        >
          <MoreVertIcon color="white" />
        </IconButton>

        <Menu
          open={this.state.open}
          anchorEl={this.state.anchorEl}
          onRequestClose={this.handleRequestClose}
          anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
          targetOrigin={{ horizontal: 'right', vertical: 'top' }}
        >
          <MenuItem onClick={logout}>Se d√©connecter</MenuItem>
        </Menu>
      </div>
    )
  }
}

const styles = {
  username: {
    color: 'white',
    fontSize: '18px',

    '@media screen and (max-width: 640px)': {
      display: 'none',
    },
  },

  user: {
    display: 'flex',
    alignItems: 'center',
  },

  avatar: {
    marginRight: 10,
  },
}

export default Radium(User)
