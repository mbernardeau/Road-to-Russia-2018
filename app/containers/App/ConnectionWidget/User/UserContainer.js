import { connect } from 'react-redux';
import { firebaseConnect } from 'react-redux-firebase';
import { compose } from 'redux';

import User from './User';

export default compose(
  firebaseConnect(),
  connect(
    ({ firebase: { profile } }) => ({
      user: profile,
    })
  )
)(User);

