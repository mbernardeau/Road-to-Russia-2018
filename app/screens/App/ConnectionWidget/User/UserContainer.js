import { connect } from 'react-redux'
import { firebaseConnect } from 'react-redux-firebase'
import { compose } from 'redux'
import betFactory from 'redux/bets'
import { getProfile } from 'redux/user'

import User from './User'

const mapState = state => ({
  user: getProfile(state),
})

const mapDispatch = (dispatch, { firebase: { logout } }) => ({
  logout: () => {
    logout()
    dispatch(betFactory.reset())
  },
})

export default compose(firebaseConnect(), connect(mapState, mapDispatch))(User)
