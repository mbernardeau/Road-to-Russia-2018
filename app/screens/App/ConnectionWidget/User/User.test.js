import React from 'react'
import renderer from 'react-test-renderer'
import Radium from 'radium'
import User from './User'

describe('screens/ConnectionWidget/User', () => {
  Radium.TestMode.enable()

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <User
          user={{ avatarUrl: 'http://avatar.url', displayName: 'Georges Lucas' }}
          firebase={{ logout: () => {} }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
