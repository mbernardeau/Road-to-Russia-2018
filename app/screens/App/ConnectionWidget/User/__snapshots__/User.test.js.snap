// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/ConnectionWidget/User renders correctly 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <div
    className="MuiAvatar-root-1"
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <img
      alt={undefined}
      className="MuiAvatar-img-3"
      sizes={undefined}
      src="http://avatar.url"
      srcSet={undefined}
    />
  </div>
  <span
    className="rmq-4ea42308"
    data-radium={true}
    style={
      Object {
        "color": "white",
        "fontSize": "18px",
      }
    }
  >
    Georges Lucas
  </span>
  <button
    aria-haspopup="true"
    aria-label="Plus"
    aria-owns={null}
    className="MuiButtonBase-root-13 MuiIconButton-root-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label-10"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-15 MuiIconButton-icon-11"
        color="white"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-16"
    />
  </button>
</div>
`;
